Filter/AbstractDateFilter.php

```diff
@@ -165,6 +165,24 @@ public function getRenderSettings()
             $name .= '_range';
         }
 
+        // NEXT_MAJOR: Remove this line when drop Symfony <2.8 support
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+            switch ($name) {
+                case 'sonata_type_filter_date':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateType';
+                    break;
+                case 'sonata_type_filter_date_range':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateRangeType';
+                    break;
+                case 'sonata_type_filter_datetime':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateTimeType';
+                    break;
+                case 'sonata_type_filter_datetime_range':
+                    $name = 'Sonata\AdminBundle\Form\Type\Filter\DateTimeRangeType';
+                    break;
+            }
+        }
+
```

Comment:
There are classes to convert snake case to camel case and so on. Doctrine inflector or maybe sf has  its own class